docs/source/dev-guide/deep-dive-install.md

```diff
@@ -0,0 +1,646 @@
+    * Linking (adding a package to the environment)
+    * Compiling bytecode (generating the `pyc` counterpart for each `py` module)
+    * Adding entry points (generate command line executables for the configured functions)
+    * Adding the JSON records (for each package, a JSON file is added to `conda-meta/`)
+    * Make menu items (create shortcuts for packages featuring a JSON file under `Menu/`)
+    * Remove menu items (remove the shortcuts created by that package)
+
+It's important to notice that download and extraction happen separately from all the other actions.
+This separation is important and core to the idea of what a `conda` environment is. Essentially,
+when you create a new `conda` environment, you are not necessarily _copying_ files over to the target
+prefix location. Instead, `conda` maintains a cache of every package ever downloaded to disk (both
+the tarball and the extracted contents). To save space and speed up environment creation and
+deletion, files are not copied over, but instead they are linked (usually via a hardlink). That's
+why these two tasks are separated in the transaction logic: you don't need to download and extract
+packages that are already in the cache; you only need to link them!
+
+````{admonition} Transactions also drive reports
+
+The type and number of actions can also be calculated by `_make_legacy_action_groups()`, which
+returns a list of _action groups_ (one per `PrefixSetup`). Each action group is a just a dictionary
+following this specification:
+
+```
+{
+  "FETCH": Iterable[PackageRecord],  # estimated by `ProgressiveFetchExtract`
+  "PREFIX": str,
+  "UNLINK": Iterable[PackageRecord],
+  "LINK: Iterable[PackageRecord],
+}
+```
+
+These simpler action groups are only used for reporting, either via a processed text report
+(via `print_transaction_summary`) or just the raw JSON (via `stdout_json_success`). As you can see,
+they do not know anything about other types of tasks.
+````
+
+## Download and extraction
+
+`conda` maintains a cache of downloaded tarballs and their extracted contents to save disk space
+and improve the performance of environment modifications. This requires some code to check whether
+a given `PackageRecord` is already present in the cache, and, if it's not, how to download the
+tarball and extract its contents in a performant way. This is all handled by the
+`ProgressiveFetchExtract` class, which can instantiate up to two `Action` objects for each
+passed `PackageRecord`:
+
+* `CacheUrlAction`: downloads (if remote) or copies (if local) a tarball to the cache location.
+* `ExtractPackageAction`: extracts the contents of the tarball.
+
+These two actions only take place _if_ the package is not in cache yet and if it has already been
+extracted, respectively. They can also revert the changes if the transaction is aborted (either
+due to an error or because the user pressed Ctrl+C).
+
+```{admonition} Anatomy of the conda caches
+TODO
+```
```

Comment:
Also a TODO?

Reply:
I commented this out for now, not crucial to understanding the whole thing.